src/plugins/console/public/application/hooks/use_data_init/use_data_init.test.ts

```diff
@@ -0,0 +1,201 @@
+        text: {
+          findAll: jest.fn(() => (typeof arg === 'function' ? arg() : arg)),
+          create: jest.fn((data: any) => mockedObject),
+        },
+      },
+    },
+  });
+
+  beforeAll(() => {
+    setStateMock = jest.fn();
+    useStateMock = (state: any) => [state, setStateMock];
+    mockContextValue = serviceContextMock.create();
+    dispatch = jest.fn();
+    // migrateToTextObjects = jest.fn();
+    (useEditorActionContext as jest.Mock).mockReturnValue(dispatch);
+    (useServicesContext as jest.Mock).mockReturnValue({
+      ...mockContextValue,
+      services: {
+        ...mockContextValue.services,
+        objectStorageClient: {
+          text: {
+            findAll: jest.fn(() => []),
+            create: jest.fn((data: any) => mockedObject),
+          },
+        },
+      },
+    });
+    // (migrateToTextObjects as jest.Mock).mockImplementation()
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  afterAll(() => {
+    jest.resetAllMocks();
+  });
+
+  it('should calls dispatch with new objects if no texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockedObject,
+    });
+  });
+
+  it('should calls dispatch with new objects if texts are provided', async () => {
+    jest.spyOn(React, 'useState').mockImplementation(useStateMock);
+
+    const mockObj: MockObjectData = {
+      createdAt: 1643277939899,
+      updatedAt: 1643277939900,
+      text: 'Mocked data',
+    };
+
+    (useServicesContext as jest.Mock).mockReturnValue(callMockFuncWithArg([mockObj]));
+
+    renderHook(() => useDataInit(), {});
+
+    await wait(0);
+
+    expect(dispatch).toHaveBeenCalledWith({
+      type: 'setCurrentTextObject',
+      payload: mockObj,
+    });
+  });
+
+  it('should update the states if the retry function is called', () => {
+    const setErrorMock = jest.fn();
+    const useErrorMock: UseStateMock = (state: any) => [state, setErrorMock];
+
+    const setDoneMock = jest.fn();
+    const useDoneMock: UseStateMock = (state: any) => [state, setDoneMock];
+
+    const setRetryTokenMock = jest.fn();
+    const useRetryTokenMock: UseStateMock = (state: any) => [state, setRetryTokenMock];
+
+    jest
+      .spyOn(React, 'useState')
+      .mockImplementationOnce(useErrorMock)
+      .mockImplementationOnce(useDoneMock)
+      .mockImplementationOnce(useRetryTokenMock);
+
+    const { result } = renderHook(() => useDataInit(), {});
+
+    result.current.retry();
+
+    expect(setErrorMock).toHaveBeenCalledWith(null);
+    expect(setDoneMock).toHaveBeenCalledWith(false);
+    expect(setRetryTokenMock).toHaveBeenCalledWith({});
+  });
+
+  it('should update the error state if case of getting an exception while loading data', async () => {
```

Comment:
`should update the error state if case of getting an exception while loading data` -> `should update the error state in case of getting an exception while loading data`